services/QuillLMS/spec/models/app_setting_spec.rb

```diff
@@ -0,0 +1,109 @@
+# == Schema Information
+#
+# Table name: app_settings
+#
+#  id                 :integer          not null, primary key
+#  enabled            :boolean          default(FALSE), not null
+#  enabled_for_admins :boolean          default(FALSE), not null
+#  name               :string           not null
+#  pct_active         :integer          default(0), not null
+#  user_whitelist     :string           default([]), not null, is an Array
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# Indexes
+#
+#  index_app_settings_on_name  (name)
+#
+require 'rails_helper'
+
+RSpec.describe AppSetting, type: :model do
+
+  describe '#enabled?' do 
+    let(:user) { create(:user) }
+    let!(:app_setting) { create(:app_setting, name: 'egosum') }
+
+    context 'AppSetting does not exist' do
+      it 'should return false' do 
+        expect(AppSetting.enabled?(name: 'nemo', user: user)).to be false
+      end
+    end
+
+    context 'User does not exist' do 
+      it 'should return false' do 
+        expect(AppSetting.enabled?(name: 'egosum', user: nil)).to be false
+      end
+    end
+
+    context 'User and AppSetting exist' do 
+      it 'should invoke _enabled?' do 
+        expect_any_instance_of(AppSetting).to receive(:_enabled?).once
+        AppSetting.enabled?(name: 'egosum', user: user)
+      end
+    end
+  end
+
+  describe '#_enabled?' do 
+    
+    context 'override is false' do 
+      let(:as1) { create(:app_setting, pct_active: 40, enabled: false) }
+
+      it 'should return false' do 
+        u1 = create(:user)
+        expect(as1._enabled?(u1)).to be false
+      end      
+
+    end
+
+    context 'override is true, admin is true' do 
+      let(:as1) { create(:app_setting, pct_active: 10, enabled: true, enabled_for_admins: true) }
+
+      it 'should return true when user is admin' do 
+        allow_any_instance_of(User).to receive(:admin?) { true }
+        u1 = create(:user, id: 9)
+        expect(as1._enabled?(u1)).to be true
+      end
+    
+      it 'should return true for non-admin user when user is in whitelist' do 
+        allow_any_instance_of(User).to receive(:admin?) { false } 
+        as1.user_whitelist = [9]
+        u1 = create(:user, id: 9)
```

Comment:
I think that it might be possible that somehow a `User` with id 9 could exist in the db based on earlier tests.  Probably safer to do.
```ruby
u1 = create(:user)
as1.user_whitelist[u1.id]
```